File: bench/interp/bench_dict.py
import time
n = 200000
keys = ["key%d" % i for i in range(n)]
t0 = time.perf_counter()
m = {}
for i in range(n): m[keys[i]] = i
total = 0
for _ in range(3):
    for i in range(n): total += m[keys[i]]
print("result: %d" % total)
print("elapsed: %d ms" % int((time.perf_counter()-t0)*1000))

File: bench/interp/bench_dict.lua
-- bench_dict.lua - string-keyed map build + lookup
local n = 200000
local keys = {}
for i = 0, n-1 do keys[i] = "key" .. i end
local t0 = os.clock()
local m = {}
for i = 0, n-1 do m[keys[i]] = i end
local total = 0
for _ = 1, 3 do
    for i = 0, n-1 do total = total + m[keys[i]] end
end
local elapsed = math.floor((os.clock() - t0) * 1000)
print("result: " .. total)
print("elapsed: " .. elapsed .. " ms")
